schemas/openshift/terraform-resource-1.yml

```diff
@@ -794,8 +788,27 @@ oneOf:
       properties:
         id:
           type: string
-      required:
-      - id
+        repo:
+          type: object
+          additionalProperties: false
+          properties:
+            tag_name:
+              type: string
+            url:
+              type: string
+              format: uri
+            ref:
+              type: string
+              pattern: '^([0-9a-f]{40}|master|main)$'
+          required:
+          - tag_name
```

Comment:
i understand ref and url, but what is tag_name?

Reply:
We assume AMI should be tagged with a tag_name and its value should be commit_sha comes from url and ref. So we can filter AMI with that tag. tag_name can be customized by tenants(e.g image builder use composer_commit)